Assets/Scripts/Achievements/SteamAchievementManager.cs

```diff
@@ -1,33 +1,62 @@
-using System.Collections;
 using System.Collections.Generic;
-using UnityEngine;
 using Steamworks;
-using Utilities;
+using UnityEngine;
 
 namespace Achievements
 {
-    public class SteamAchievementManager
+    public static class SteamAchievementManager
     {
-        // Corresponds to the enum values in Enums.cs
-        private List<string> achievementNames = new List<string>()
-        {
-            "TEST_0", 
-            "BUILD_1_BUILDING"
-        };
+        public static bool Initialised => SteamManager.Initialized;
+
+        private static readonly Dictionary<GameStat, string> StatIDs = 
+            new Dictionary<GameStat, string>
+            {
+                {GameStat.Population, "GREATEST_POPULATION"},
+                {GameStat.Turn, "GREATEST_TURN"}
+            };
 
-        public void SetOneTimeAchievement(Achievement achievement)
+        private static readonly Dictionary<Achievement, string> AchievementIDs = 
+            new Dictionary<Achievement, string>
+            {
+                {Achievement.PetDog, "PET_THE_DOG"},
+                {Achievement.BuildOneBuilding, "BUILD_FIRST_BUILDING"},
+                {Achievement.PopulationVillage, "POPULATION_VILLAGE"},
+                {Achievement.TurnMonth, "TURN_MONTH"}
+            };
+
+        public static void Unlock(Achievement achievement)
         {
-            if (IsSteamWorksInitialised()) SteamUserStats.SetAchievement(achievementNames[(int) achievement]);
+            if (!Initialised) return;
+            SteamUserStats.SetAchievement(AchievementIDs[achievement]);
+            SteamUserStats.StoreStats();
         }
-
-        bool IsSteamWorksInitialised()
+        
+        public static void Update(GameStat stat, int value)
+        {
+            if (!Initialised) return; 
+            SteamUserStats.SetStat(StatIDs[stat], value);
+            SteamUserStats.StoreStats();
+        }
+        
+        public static void UpdateProgress(GameStat stat, Achievement achievement, int value)
         {
-            return SteamManager.Initialized;
+            if (!Initialised) return;
+            // var achievementID = AchievementIDs[achievement];
+            
+            // TODO  ISteamUserStats_GetAchievementProgressLimitsInt32 fails... why??
+            // var ret = SteamUserStats.GetAchievementProgressLimits(achievementID, out _, out int max);
+            // SteamUserStats.GetStat(StatIDs[stat], out int upstreamStat);
+            // Debug.Log($"progress for {achievementID} {value} of {max} with upstream {upstreamStat} status {ret}");
+            //
+            // if (value < upstreamStat || value >= max) return;
+            // SteamUserStats.IndicateAchievementProgress(achievementID, (uint) value, (uint) max);
         }
```

Comment:
Spent most of my time trying to figure out why this one here broke. `SteamUserStats.GetAchievementProgressLimits` keeps failing and I have no idea why. The internet was not helpful, but this is to essentially show progress notifications to the user.

Link to docs: https://partner.steamgames.com/doc/api/ISteamUserStats